### PR TITLE
COZMO-12173 Quizmaster Cozmo example

### PR DESCRIPTION
**Pull request by MarkWesleyAtAnki**
Add an example quiz program using the cubes as buzzers
Add Json files to the examples zip (as quiz uses one)
Updated run.py to more cleanly handle exit of async programs when the user presses CTRL-C to keyboard interrupt
Added objects.LightCubeIDs helper list of all 3 cube IDs

### CODE REVIEW

File: examples/apps/quizmaster_cozmo.py [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+        if len(options) != 4:
+            raise ValueError("Expected 4 options, got %s" % len(options))
+        self.question = question
+        self._answer_index = 0
+        self.options = list(options)  # copy the options, so we can shuffle them
+        self.shuffle_options()
+
+    @property
+    def answer_number(self):
+        """int: The number (1..4) representing the correct answer."""
```
Comment (msintov) [inline examples/apps/quizmaster_cozmo.py]: To describe the list in a way all people should understand, could do: (i.e., 1, 2, 3 or 4). I know in the SDK we already describe lists like (1..4), but for a higher level user who might want to use this example, instead I think we should use a more typical description.

File: examples/apps/quizmaster_cozmo.py [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+        return self._answer_index + 1
+
+    @property
+    def answer_str(self):
+        """str: The string representing the correct answer."""
+        return self.options[self._answer_index]
+
+    def shuffle_options(self):
```
Comment (msintov) [inline examples/apps/quizmaster_cozmo.py]: Again, could rename to shuffle_answer_options or similar.

File: examples/apps/quizmaster_cozmo.py [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+        """Shuffle the options so that they're not always read in the same order."""
+        # to shuffle whilst keeping track of the answer, we first pop the
```
Comment (msintov) [inline examples/apps/quizmaster_cozmo.py]: Could add a line of whitespace above the # comment line so it's more readable.

File: examples/apps/quizmaster_cozmo.py [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+        self.options.insert(self._answer_index, answer)
+
+
+class CozmoQuizPlayer:
+    """A player in the quiz.
+
+    Args:
+        robot (:class:`cozmo.robot.Robot`): The cozmo robot.
+        cube (:class:`cozmo.objects.LightCube`): This player's cube.
+        index(int): The number (0..2) specifying the index of this player and cube.
```
Comment (msintov) [inline examples/apps/quizmaster_cozmo.py]: Again, could write the list instead as "(i.e., 0, 1, or 2)"

File: examples/apps/quizmaster_cozmo.py [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+        self.name = name
+        self.score = 0
+        self._has_buzzed_in = False
+        self._answer_index = 0
+
+    def verify_setup(self):
+        # Return True if and only if the player was setup correctly and has a connected cube.
+        success = True
+        if self._cube is None:
+            cozmo.logger.warning("Cozmo is not connected to a cube %s - check the battery.", (self._index+1))
```
Comment (msintov) [inline examples/apps/quizmaster_cozmo.py]: Is a battery failure the most common reason the value would be None?

File: examples/apps/quizmaster_cozmo.py [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+                conjunction = " or " if is_last_option else ", "
+            text += conjunction + str(i+1) + ": " + list_of_options[i]
+        return text
+
+    def say_text(self, text, in_parallel=False):
+        print("%s" % text)
+        return self._robot.say_text(text, in_parallel=in_parallel)
+
+    async def wait_for_answer(self, player):
+        # Wait for player's answer (whatever they leave selected after x seconds)
```
Comment (msintov) [inline examples/apps/quizmaster_cozmo.py]: "they" is confusing here. Could say "whatever the player leaves selected..."

File: examples/apps/quizmaster_cozmo.py [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+        winning_players = []
+        for player in self._players:
+            if len(winning_players) == 0 or player.score > winning_players[0].score:
+                winning_players = [player]
+            elif player.score == winning_players[0].score:
+                winning_players.append(player)
+        return winning_players
+
+    async def report_leader(self, is_final_score):
+        # report however is currently winning / won
```
Comment (msintov) [inline examples/apps/quizmaster_cozmo.py]: Could rewrite as "Report the winning player" or similar. "however" should be "whoever" if original text is kept.

File: examples/apps/quizmaster_cozmo.py [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+            else:
+                # Incorrect
+                player.score -= 1
+                action = self._robot.play_anim_trigger(cozmo.anim.Triggers.KnockOverFailure)
+                self._answering_player = None
+                player.turn_light_off()
+                await action.wait_for_completed()
+
+    async def ask_question(self, question: QuizQuestion):
+        # Reset everyone for the question
```
Comment (msintov) [inline examples/apps/quizmaster_cozmo.py]: Comment might be clearer as "Reset for a new question"

File: examples/apps/quizmaster_cozmo.py (hunk 5 above)
Comment (MarkWesleyAtAnki) [inline examples/apps/quizmaster_cozmo.py]: i think so - generally it's because either the cube isn't there (e.g. they lost it) or it's battery is dead

**Pull request merged**